source/core/CEntryPoints.cpp

```diff
@@ -501,10 +501,28 @@ VIREO_EXPORT Int32 TypeRef_Alignment(TypeRef typeRef)
     return typeRef->AQAlignment();
 }
 //------------------------------------------------------------
-VIREO_EXPORT void TypeRef_Name(TypeRef typeRef, Int32* bufferSize, char* buffer)
+VIREO_EXPORT const char* TypeRef_Name(TypeRef typeRef)
 {
     SubString name = typeRef->Name();
-    *bufferSize = name.CopyToBoundedBuffer(*bufferSize, reinterpret_cast<Utf8Char*>(buffer));
+
+    static StringRef returnBuffer = nullptr;
+    if (returnBuffer == nullptr) {
```

Comment:
Cool, I'll add it.

Reply:
I don't think we need the TypeManagerScope in this situation. We already have the TypeRef. Most functions use the scope when a TypeManagerRef is passed.

Reply:
Don't we need it because of the STACK_VAR allocation? It has a check for the TypeManagerScope, so it seemed like we should be setting the scope to the same as other files

Reply:
Talked to Carlos and agrees that we need the TympeManager ref and scope call to allocate the string with STACK_VAR